### ISSUE TITLE
Build is broken on master: function or associated item not found in `proc_macro2::Literal`

### ISSUE THREAD

**Issue by alamb** [apache/arrow-datafusion#1498]
**Describe the bug**

After `cargo updating` the build is broken:

```
alamb@MacBook-Pro-2 arrow-datafusion % cargo update
cargo update
    Updating crates.io index
    Updating anyhow v1.0.51 -> v1.0.52
    Updating quote v1.0.10 -> v1.0.11
    Updating typenum v1.14.0 -> v1.15.0
    Updating zstd v0.9.0+zstd.1.5.0 -> v0.9.1+zstd.1.5.1
    Updating zstd-safe v4.1.1+zstd.1.5.0 -> v4.1.2+zstd.1.5.1
    Updating zstd-sys v1.6.1+zstd.1.5.0 -> v1.6.2+zstd.1.5.1
alamb@MacBook-Pro-2 arrow-datafusion %
```

```
error[E0599]: no function or associated item named `from_str_unchecked` found for struct `proc_macro2::Literal` in the current scope
   --> /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.11/src/runtime.rs:301:41
    |
301 |         let literal = unsafe { Literal::from_str_unchecked(repr) };
    |                                         ^^^^^^^^^^^^^^^^^^ function or associated item not found in `proc_macro2::Literal`

error[E0599]: no function or associated item named `from_str_unchecked` found for struct `proc_macro2::Literal` in the current scope
   --> /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.11/src/runtime.rs:310:45
    |
310 |         let mut literal = unsafe { Literal::from_str_unchecked(repr) };
    |                                             ^^^^^^^^^^^^^^^^^^ function or associated item not found in `proc_macro2::Literal`
```

Also happens on CI: https://github.com/apache/arrow-datafusion/runs/4649830794?check_suite_focus=true

**To Reproduce**

**Expected behavior**

**Additional context**

**Comment by alamb** [comment]
I found a workaround of doing the following forcing the `quote` package back to `1.0.10` (from `1.0.11`)

```rust
cargo update --package quote --precise 1.0.10
```

**Comment by alamb** [comment]
Seems it was fixed in `quote` 1.0.12 

https://github.com/dtolnay/quote/releases/tag/1.0.12

However I can't update it for some reason: 

```
alamb@MacBook-Pro-2 arrow-datafusion % cargo update --package quote --precise 1.0.12
cargo update --package quote --precise 1.0.12
    Updating crates.io index
error: failed to select a version for `proc-macro2`.
    ... required by package `quote v1.0.12`
    ... which satisfies dependency `quote = "^1.0"` of package `async-trait v0.1.52`
    ... which satisfies dependency `async-trait = "^0.1.36"` (locked to 0.1.52) of package `ballista-core v0.6.0 (/Users/alamb/Software/arrow-datafusion/ballista/rust/core)`
    ... which satisfies path dependency `ballista-core` (locked to 0.6.0) of package `arrow-benchmarks v5.0.0 (/Users/alamb/Software/arrow-datafusion/benchmarks)`
versions that meet the requirements `^1.0.36` are: 1.0.36

all possible versions conflict with previously selected packages.

  previously selected package `proc-macro2 v1.0.27`
    ... which satisfies dependency `proc-macro2 = "=1.0.27"` of package `arrow-flight v6.4.0`
    ... which satisfies dependency `arrow-flight = "^6.4.0"` (locked to 6.4.0) of package `ballista-core v0.6.0 (/Users/alamb/Software/arrow-datafusion/ballista/rust/core)`
    ... which satisfies path dependency `ballista-core` (locked to 0.6.0) of package `arrow-benchmarks v5.0.0 (/Users/alamb/Software/arrow-datafusion/benchmarks)`

failed to select a version for `proc-macro2` which could resolve this conflict
```
